config/metadata/rulesMeta.yaml

```
@@ -651,6 +651,11 @@ groups:
         summary: is the comparison operator to use.
         description: >
           The comparison operator to use. String comparisons are case-sensitive.
+          WARNING: using the `not-exists` operator in a rule with a scope of
+          "trace" (the default) will cause the rule to match a trace if *any* span
+          in the trace does not have the field. This is almost certainly not the
+          matching behavior you want. Use `not-exists` in a rule with a scope of
+          "span".
```

```suggestion
          The comparison operator to use. String comparisons are case-sensitive.
          For most cases, use `not-exists` in a rule with a scope of "span".
          WARNING: using the `not-exists` operator in a rule with a scope of
          "trace" (the default) will cause the rule to match a trace if *any* span
          in the trace does not have the field. This is almost certainly not the
          matching behavior you want. 
```

I think this could be worded better, but the general suggestion is to put the desired behavior first and then outline the warning behavior.

---

I have reworded to "do this" first with a Frankenstein's monster of my original, Kent's suggestion, and your suggestion